src/elements/content-explorer/ContentExplorer.js

```diff
@@ -725,6 +729,38 @@ class ContentExplorer extends Component<Props, State> {
         }
     };
 
+    /**
+     * Returns an object with newCollection and newSelected properties.
+     * newCollection will be the given collection, with items.selected properties
+     * updated according to the given selected param. newSelected will be the selected
+     * item if it is in newCollection, otherwise undefined.
+     *
+     * @private
+     * @param {Collection} collection - collection that needs to be updated
+     * @param {?BoxItem} selected - item that should be selected in that collection (if present)
+     * @return {Object}
+     */
+    updateCollection(collection: Collection, selected: ?BoxItem): Object {
+        const newCollection: Collection = { ...collection };
+        const targetID = selected ? selected.id : null;
+
+        const ret = { newCollection, newSelected: undefined };
+
+        if (collection.items) {
+            ret.newCollection.items = collection.items.map(obj => {
+                const item = { ...obj, selected: obj.id === targetID };
```

Comment:
It seems like we're tracking the same piece of information, `selected`,  twice: once in the collection and once in a top-level state property. Consolidating to one or the other seems like it would simplify things a bit.